src/kedro_vizro_labeling/dashboards/protected_action.py

```diff
@@ -173,6 +175,9 @@ def build(self) -> html.Div:
         def callback_wrapper(external: list[Any] | dict[str, Any], internal: dict[str, Any]) -> dict[str, Any]:
             unallowed_output = (no_update, ) * self._n_outputs
 
+            # AM comment: I think there's probably better ways to inject the extra outputs without overriding so much
+            # here and in the other methods. Let me try and think of how else to do it.
```

Comment:
I have seen the way you use `set_props` and it looks really great. Is it what you are referring to?

Reply:
Actually that's a method I only thought of after writing this comment - at the time I was thinking you could do something similar to what you do but without overriding so many methods in `ProtectedAction`, e.g. I think you can probably do it without overriding `build`. I was about to write a version of this and then thought of the `set_props` method so tried that one out instead. Overall, either the `set_props` method or a less-overriding method seem like good options I think.

Reply:
I'll give it a try, thanks! :)

Reply:
Used `set_props` to significantly simplify `ProtectedAction` :)